.evergreen/config.yml

```diff
@@ -1,48 +1,191 @@
+# Mark a failure as a system/bootstrap failure (purple box) rather then a task
+# failure by default.
+# Actual testing tasks are marked with `type: test`
+command_type: system
+
+# Protect ourselves against rogue test case, or curl gone wild, that runs forever
+exec_timeout_secs: 3600
+
+# When a task that used to pass starts to fail, go through all versions that may have been skipped
+# to detect when the task started failing
+stepback: true
+
+#######################################
+#              Functions              #
+#######################################
 functions:
-  fetch source:
-    # Executes git clone
+  "fetch source":
+    # Executes git clone and applies the submitted patch, if any.
     - command: git.get_project
       params:
         directory: src
-    # Applies the submitted patch, if any
-    - command: git.apply_patch
 
-  exec script:
-    - command: shell.exec
-      type: test
+  "create expansions":
+    - command: subprocess.exec
+      params:
+        working_dir: src
+        binary: bash
+        args:
+          - .evergreen/create-expansions.sh
+        include_expansions_in_env:
+          - is_patch
+          - version_id
+          - project
+    - command: expansions.update
+      params:
+        file: src/expansion.yml
+
+  "prepare resources":
+    - command: subprocess.exec
       params:
         working_dir: src
-        shell: bash
-        add_to_path:
-          - .evergreen
+        binary: bash
+        args:
+          - .evergreen/fetch-drivers-tools.sh
+        include_expansions_in_env:
+          - DRIVERS_TOOLS
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          echo "{ \"releases\": { \"default\": \"$MONGODB_BINARIES\" }}" > $MONGO_ORCHESTRATION_HOME/orchestration.config
+
+  "bootstrap mongo-orchestration":
```

Comment:
I hesitated about such convention to use '-' as word separator for the following two reasons:

- `mongo-orchestration` is the naming convention for this component so we could treat it as one single name instead
- if we use `-` as separator, it would conflict with the above `mongo-orchestration`.

My preference would be to stick to the current naming convention by using space as word separator, but the above `mongo-orchestration` is special in that that is usually the name for it.

Reply:
C# might be special in that C# has special naming convention as one unique language feature (e.g. see https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names). I would suggest being cautious about adopting its idiocracy.

Reply:
I saw we didn't use separator in `stop mongo orchestration`. So I deleted the separator here to ensure consistency.

Reply:
We decided to use dashes everywhere, for all new code. No quotes.